python/scope.py

```diff
@@ -2060,6 +2064,33 @@ def type_check_command(self, tree):
             self.variables[name.value] = Variable(
                 constructor_type, constructor_type, public
             )
+
+            scope = self.new_scope(parent_function=None)
+            for arg_pattern, arg_type in arguments:
+                scope.assign_to_pattern(arg_pattern, arg_type, True, certain=True)
+            scope.type_check_command(class_body)
+
+            for prop_name, var in scope.variables.items():
+                if var.public:
+                    if var.type is None:
+                        class_type = "invalid"
+                        break
+                    else:
+                        class_type[prop_name] = var.type
+
+            self.types[name.value] = class_type
+            if public:
+                self.public_types[name.value] = self.types[name.value]
+
+            constructor_type = tuple(
+                [*(arg_type for _, arg_type in arguments), class_type]
+            )
+            self.variables[name.value] = Variable(
+                constructor_type, constructor_type, public
+            )
```

Comment:
You should move this before the first `scope.type_check_command(class_body)` so that Test is defined